Sender from msg can't be used for MimeMessage

I get an an exception when I try to create a MimeMessage from a OutlookMessage:
```
Caused by: javax.mail.internet.AddressException: Local address contains control or whitespace in string ``Tim Toner''
	at javax.mail.internet.InternetAddress.checkAddress(InternetAddress.java:1353)
	at javax.mail.internet.InternetAddress.parse(InternetAddress.java:1213)
	at javax.mail.internet.InternetAddress.parse(InternetAddress.java:752)
	at javax.mail.internet.InternetAddress.<init>(InternetAddress.java:119)
	at ch.astorm.jotlmsg.OutlookMessage.toMimeMessage(OutlookMessage.java:371)
	at ch.astorm.jotlmsg.OutlookMessage.toMimeMessage(OutlookMessage.java:352)
	at ch.astorm.jotlmsg.OutlookMessage.toMimeMessage(OutlookMessage.java:339)
	at de.adito.msgconverter.Msg2eml.<init>(Msg2eml.java:31)
```

I were able to find the sender E-Mail address in the 'unknown chunks' in 'org.apache.poi.hsmf.MAPIMessage' but the actual field for the address is empty. Only the name is supplied.

--- a/src/main/java/ch/astorm/jotlmsg/OutlookMessage.java
+++ b/src/main/java/ch/astorm/jotlmsg/OutlookMessage.java
@@ -56,6 +56,8 @@ import java.util.EnumMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import javax.activation.DataHandler;
 import javax.mail.Address;
 import javax.mail.MessagingException;
@@ -368,14 +370,15 @@ public class OutlookMessage {
         String subject = getSubject();
         if(subject!=null) { message.setSubject(subject); }
         
-        String from = getFrom();
+        String from = extractEmail(getFrom());
         if(from!=null) { message.setFrom(new InternetAddress(from)); }
         
         List<String> replyTo = getReplyTo();
         if(replyTo!=null) {
             List<Address> replyAddresses = new ArrayList<>(replyTo.size());
             for(String replyToEmail : replyTo) {
-                if(replyToEmail!=null) { replyAddresses.add(new InternetAddress(replyToEmail)); }
+                String replyToEmailExtracted = extractEmail(replyToEmail);
+                if(replyToEmailExtracted!=null) { replyAddresses.add(new InternetAddress(replyToEmailExtracted)); }
             }
             message.setReplyTo(replyAddresses.toArray(new Address[replyAddresses.size()]));
         }
@@ -407,7 +410,28 @@ public class OutlookMessage {
         message.setContent(multipart);
         return message;
     }
-    
+
+    private static final Pattern MIXED_MAIL = Pattern.compile("[^\\s<>,/]+@[^\\s<>,/]+");
+
+    /**
+     * Extracts the mail from a mixed mail string, for instance "John Doe &lt;john.doe@hotmail.com&gt;".
+     * <p>This method uses a simple pattern to get the first string which looks like an email,
+     * eg there is an @ with at least one char on each side. Spaces are considered as separators.</p>
+     * <p>If there are multiple mails in {@code mixedMailStr}, only the first one will be returned.</p>
+     * <p>This method does not check at all if the returned mail is valid.</p>
+     *
+     * @param mixedMailStr A string that contains (potentially) an email address.
+     * @return The first string looking like an email or null.
+     */
+    public static String extractEmail(String mixedMailStr) {
+        if(mixedMailStr==null || mixedMailStr.trim().isEmpty()) {
+            return null;
+        }
+
+        Matcher matcher = MIXED_MAIL.matcher(mixedMailStr);
+        return matcher.find() ? matcher.group() : null;
+    }
+
     /**
      * Writes the content of this message to the specified {@code file}. The created
      * file will be in format {@code .msg} that can be open by Microsoft Outlook.

--- a/src/test/java/ch/astorm/jotlmsg/OutlookMessageTest.java
+++ b/src/test/java/ch/astorm/jotlmsg/OutlookMessageTest.java
@@ -101,4 +101,24 @@ public class OutlookMessageTest {
         String data = IOUtils.toString(message.getAttachments().get(0).getNewInputStream(), "UTF-8");
         assertEquals("This is some basic content of attached file.", data);
     }
+
+    @Test
+    public void testEmailExtraction() {
+        assertNull(OutlookMessage.extractEmail(null));
+        assertNull(OutlookMessage.extractEmail("      "));
+        assertNull(OutlookMessage.extractEmail("John"));
+        assertNull(OutlookMessage.extractEmail("John Doe"));
+        assertNull(OutlookMessage.extractEmail("John Doe@ gmail"));
+        assertNull(OutlookMessage.extractEmail("@John"));
+        assertNull(OutlookMessage.extractEmail("John @Doe"));
+        assertEquals("john.doe@gmail.com", OutlookMessage.extractEmail("john.doe@gmail.com"));
+        assertEquals("john.doe@gmail.com", OutlookMessage.extractEmail("John Doe <john.doe@gmail.com>"));
+        assertEquals("john.doe@gmail.com", OutlookMessage.extractEmail("<john.doe@gmail.com>"));
+        assertEquals("john_doe+someTag@gmail.com", OutlookMessage.extractEmail("john_doe+someTag@gmail.com"));
+        assertEquals("john.doe_extended+someTag@gmail.com", OutlookMessage.extractEmail("John Doe <john.doe_extended+someTag@gmail.com>"));
+
+        assertEquals("john.doe@gmail.com", OutlookMessage.extractEmail("john.doe@gmail.com,francis.max@blast.com"));
+        assertEquals("john.doe@gmail.com", OutlookMessage.extractEmail("John <john.doe@gmail.com>,francis.max@blast.com"));
+        assertEquals("john.doe@gmail.com", OutlookMessage.extractEmail("john.doe@gmail.com , Francis <francis.max@blast.com>"));
+    }
 }

